x11-libs/libice/libice-1.0.9.recipe

```diff
@@ -0,0 +1,75 @@
+SUMMARY="Inter-Client Exchange Library"
+DESCRIPTION="The ICE library utilizes callbacks to process incoming \
+messages. Using callbacks allows ProtocolSetup messages and authentication \
+to happen behind the scenes. An additional benefit is that messages never \
+need to be buffered up by the library when the client blocks waiting for \
+a particular message."
+HOMEPAGE="http://xorg.freedesktop.org/releases/individual/lib/"
+COPYRIGHT="1993, 1998 The Open Group"
+LICENSE="MIT (no promotion)"
+REVISION="1"
+SOURCE_URI="https://www.x.org/releases/individual/lib/libICE-$portVersion.tar.gz"
+CHECKSUM_SHA256="7812a824a66dd654c830d21982749b3b563d9c2dfe0b88b203cefc14a891edc0"
+SOURCE_FILENAME="libICE-$portVersion.tar.gz"
+SOURCE_DIR="libICE-$portVersion"
+
+ARCHITECTURES="!x86_gcc2 x86 ?x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+libVersion=6.3.0
+libVersionCompat="$libVersion compat >= 6"
+
+PROVIDES="
+	libice$secondaryArchSuffix = $portVersion
+	lib:libICE$secondaryArchSuffix = $libVersionCompat
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	libice${secondaryArchSuffix}_devel = $portVersion
+	devel:libICE$secondaryArchSuffix = $libVersion
+	"
+REQUIRES_devel="
+	libice$secondaryArchSuffix == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:xproto$secondaryArchSuffix
+	devel:xtrans$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	devel:util_macros$secondaryArchSuffix
+	cmd:aclocal
+	cmd:autoconf
+	cmd:libtoolize$secondaryArchSuffix
+	cmd:make
+	cmd:gcc$secondaryArchSuffix
+	cmd:pkg_config$secondaryArchSuffix
+	"
+
+defineDebugInfoPackage libice$secondaryArchSuffix \
+	$libDir/libICE.so.$libVersion
+
+BUILD()
+{
+	libtoolize --force --copy --install
+	autoreconf -i
+	runConfigure ./configure
+}
+
+INSTALL()
+{
+	make install
+
+	rm -rf $libDir/*.la
```

Comment:
No need for recursion here:
```diff
-	rm -rf $libDir/*.la
+	rm -f $libDir/*.la
```

Everything else looks OK.
Wait, you can switch HOMEPAGE to https and sort `BUILD_PREREQUIRES`.

And maybe move `devel:util_macros$secondaryArchSuffix` from `BUILD_PREREQUIRES` to `BUILD_REQUIRES`.
Unless it is there for some special reason.

(I did not test anything)

Reply:
Will do, this was based on an earlier X11 lib, so maybe there could be some moves needed too ..

Reply:
Just remembered, you need util_macros to be able to run autoreconf, so I think it should stay in there?

Reply:
I don't know. Maybe you are right.
If it works this way then I'm not against keeping  `devel:util_macros$secondaryArchSuffix` in `BUILD_PREREQUIRES` (unless someone else sheds some more light telling it should be moved.)